contracts/governance/Vesting/VestingCreator.sol

```diff
@@ -0,0 +1,286 @@
+		address tokenOwner;
+		uint256 vestingCreationType;
+	}
+
+	///@notice list of vesting to be processed
+	VestingData[] public vestingDataList;
+
+	event SOVTransferred(address indexed receiver, uint256 amount);
+	event TokensStaked(address indexed vesting, address indexed tokenOwner, uint256 amount);
+	event VestingDataRemoved(address indexed caller, address indexed tokenOwner);
+	event DataCleared(address indexed caller);
+
+	constructor(address _SOV, address _vestingRegistryProxy) public {
+		require(_SOV != address(0), "SOV address invalid");
+		require(_vestingRegistryProxy != address(0), "Vesting registry address invalid");
+
+		SOV = IERC20(_SOV);
+		vestingRegistryLogic = VestingRegistryLogic(_vestingRegistryProxy);
+	}
+
+	/**
+	 * @notice transfers SOV tokens to given address
+	 * @param _receiver the address of the SOV receiver
+	 * @param _amount the amount to be transferred
+	 */
+	function transferSOV(address _receiver, uint256 _amount) external onlyOwner {
+		require(_amount != 0, "amount invalid");
+		require(SOV.transfer(_receiver, _amount), "transfer failed");
+		emit SOVTransferred(_receiver, _amount);
+	}
+
+	/**
+	 * @notice adds vestings to be processed to the list
+	 */
+	function addVestings(
+		address[] calldata _tokenOwners,
+		uint256[] calldata _amounts,
+		uint256[] calldata _cliffs,
+		uint256[] calldata _durations,
+		bool[] calldata _governanceControls,
+		uint256[] calldata _vestingCreationTypes
+	) external onlyAuthorized {
+		require(
+			_tokenOwners.length == _amounts.length &&
+				_tokenOwners.length == _cliffs.length &&
+				_tokenOwners.length == _durations.length &&
+				_tokenOwners.length == _governanceControls.length,
+			"arrays mismatch"
+		);
+
+		for (uint256 i = 0; i < _tokenOwners.length; i++) {
+			require(_durations[i] >= _cliffs[i], "duration must be bigger than or equal to the cliff");
+			require(_amounts[i] > 0, "vesting amount cannot be 0");
+			require(_tokenOwners[i] != address(0), "token owner cannot be 0 address");
+			require(_cliffs[i].mod(TWO_WEEKS) == 0, "cliffs should have intervals of two weeks");
+			require(_durations[i].mod(TWO_WEEKS) == 0, "durations should have intervals of two weeks");
+			VestingData memory vestingData =
+				VestingData({
+					amount: _amounts[i],
+					cliff: _cliffs[i],
+					duration: _durations[i],
+					governanceControl: _governanceControls[i],
+					tokenOwner: _tokenOwners[i],
+					vestingCreationType: _vestingCreationTypes[i]
+				});
+			vestingDataList.push(vestingData);
+		}
+	}
+
+	/**
+	 * @notice Creates vesting contract and stakes tokens
+	 * @dev Vesting and Staking are merged for calls that fits the gas limit
+	 */
+	function processNextVesting() external {
+		processVestingCreation();
+		processStaking();
+	}
+
+	/**
+	 * @notice Creates vesting contract without staking any tokens
+	 * @dev Separating the Vesting and Staking to tackle Block Gas Limit
+	 */
+	function processVestingCreation() public {
```

Comment:
The idea is to add several vestings via addVestings, and then run the list backwards with processVestingCreation+processStaking calls? - **Yes**. This would be processed sequentially via backend.